tests/test_mlflow_integration.py

```diff
@@ -0,0 +1,154 @@
+import os
+import subprocess
+import sys
+
+import pandas as pd
+import pytest
+from sklearn import datasets
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+
+import mlflow
+
+
+@pytest.fixture(scope="module")
+def setup_servers():
+    with subprocess.Popen([sys.executable, "-m", "mlflow", "server", "--port", "5000"]) as proc:
+        try:
+
+            mlflow_tracking_url = "http://127.0.0.1:5000"
+            uc_oss_url = "uc:http://127.0.0.1:8080"
+
+            mlflow.set_tracking_uri(mlflow_tracking_url)
+            mlflow.set_registry_uri(uc_oss_url)
+
+            yield mlflow_tracking_url
+        finally:
+            proc.terminate()
+
+
+def test_integration(setup_servers):
+    catalog = "unity"
+    schema = "default"
+    registered_model_name = "iris"
+    model_name = f"{catalog}.{schema}.{registered_model_name}"
+    mlflow.set_experiment("iris-uc-oss")
+    try:
+        mlflow.MlflowClient().get_registered_model(model_name)
+    except Exception as e:
+        e.args[0].startswith("NOT_FOUND")
+    else:
+        assert False, "Expected exception for missing model not raised"
+
+    X, y = datasets.load_iris(return_X_y=True, as_frame=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+
+    def build_model():
+        with mlflow.start_run():
+            # Train a sklearn model on the iris dataset
+            clf = RandomForestClassifier(max_depth=7)
+            clf.fit(X_train, y_train)
+            # Take the first row of the training dataset as the model input example.
+            input_example = X.iloc[[0]]
+            # Log the model and register it as a new version in UC.
+            mlflow.sklearn.log_model(
+                clf,
+                "model",
+                # The signature is automatically inferred from the input example and
+                # its predicted output.
+                input_example=input_example,
+                registered_model_name=model_name,
+            )
+
+    try:
+        build_model()
+    except Exception as e:
+        assert False, f"Unhandled exception while attempting to build a model: {e}"
```

Comment:
Might want to just call the internal function - the stack trace should hopefully be insightful if there is a fitting error here